src/tsx/statusView/statusEffect.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+
+import { IEmpty, IEffect, EffectType } from "../types";
+import style from "../../less/statusView.module.less";
+
+interface IProps {
+  effect: IEffect;
+}
+
+class StatusEffect extends React.PureComponent<IProps, IEmpty> {
+  public render(): JSX.Element {
+    return (
+      <div className={style.effectContainer}>
+        <EffectContent effect={this.props.effect} />
+      </div>
+    );
+  }
+}
+
+function EffectContent(props: IProps) {
+  switch (props.effect.type) {
```

Comment:
I don't remember what we said about inheritance vs composition here, i.e. whether this component should use different logic depending on the effect type or (for example) just always draw a progress bar if the effect has a duration. Do you remember?

I guess we will know which approach is better once we implement more effect types though, so we could always do it like this now and refactor later if we change our mind,

Reply:
I think we might fix with inheritance later. Components may look rather different even if they have the same feature (e.g. video preview will include progress bar whilst audio may need to do a custom one).